src/Development/IDE/Plugin/CodeAction.hs

```diff
@@ -770,15 +769,29 @@ suggestFunctionConstraint contents Diagnostic{..} missingConstraint
   | Just typeSignatureName <- findTypeSignatureName _message
   = let mExistingConstraints = findExistingConstraints _message
         newConstraint = buildNewConstraints missingConstraint mExistingConstraints
-        typeSignatureLine = findTypeSignatureLine contents typeSignatureName
-        typeSignatureFirstChar = T.length $ typeSignatureName <> " :: "
-        startOfConstraint = Position typeSignatureLine typeSignatureFirstChar
-        endOfConstraint = Position typeSignatureLine $
-          typeSignatureFirstChar + maybe 0 T.length mExistingConstraints
-        range = Range startOfConstraint endOfConstraint
```

Comment:
The issue with the original Regex-based parsing of the source code was that
1. it could differ significantly in length (e.g. due to extraneous spaces of parens) compared to the "normalized" type signature in error message
2. The existing logic failed completely when there were newlines in the type signature, leading to the new Context being inserted at the end of file